pkg/operator/ceph/file/controller.go

```diff
@@ -121,10 +127,22 @@ func (c *FilesystemController) onAdd(obj interface{}) {
 			return
 		}
 	}
+	filesystem, err = updateCephFilesystemStatus(filesystem, filesystemProcessingStatus, c.context)
+	if err != nil {
+		logger.Errorf("Failed to update filesystem %s with status %v: %+v", filesystem.Name, filesystemProcessingStatus, err)
+	}
 
 	err = createFilesystem(c.clusterInfo, c.context, *filesystem, c.rookVersion, c.clusterSpec, c.filesystemOwner(filesystem), c.clusterSpec.DataDirHostPath, c.isUpgrade)
```

Comment:
If there was an err updating the status above, will the `filesystem` be nil and cause a crash here? I don't think we want to re-assign the filesystem object after that call.

Reply:
yup have made the changes. No more reassigning.